service/src/main/java/com/derpgroup/quip/bots/complibot/resource/CompliBotAlexaResource.java

```diff
@@ -81,60 +87,67 @@ public CompliBotAlexaResource(MainConfig config, Environment env) {
    * Generates a welcome message.
    *
    * @return The message, never null
-   * @throws IOException 
-   * @throws CertificateException 
-   * @throws Exception 
    */
   @POST
   public SpeechletResponseEnvelope doAlexaRequest(SpeechletRequestEnvelope request, @HeaderParam("SignatureCertChainUrl") String signatureCertChainUrl, 
-      @HeaderParam("Signature") String signature, @QueryParam("testFlag") Boolean testFlag) throws IOException, CertificateException{
-    if(testFlag == null || testFlag == false){ 
-      AlexaUtils.validateAlexaRequest(request, signatureCertChainUrl, signature);
-    }
-    if (request.getRequest() == null) {
-      throw new RuntimeException("Missing request body."); //TODO: create AlexaException
-    }
-
-    Map<String, Object> sessionAttributes = request.getSession().getAttributes();
-    sessionAttributes.put("bot", "complibot");
-
+      @HeaderParam("Signature") String signature, @QueryParam("testFlag") Boolean testFlag){
     ObjectMapper mapper = new ObjectMapper();
-    mapper.registerModule(new MixInModule());
-    QuipMetadata metadata = mapper.convertValue(sessionAttributes, new TypeReference<QuipMetadata>(){});
-    
-    SsmlDocumentBuilder builder = new SsmlDocumentBuilder(UNSUPPORTED_SSML_TAGS);
-    VoiceInput voiceInput = VoiceMessageFactory.buildInputMessage(request.getRequest(), metadata, InterfaceType.ALEXA);
-    builder.conversationEnd(false);
-    manager.handleRequest(voiceInput, builder);
-
-    Map<String,Object> sessionAttributesOutput = mapper.convertValue(metadata, new TypeReference<Map<String,Object>>(){});
-    SpeechletResponseEnvelope responseEnvelope = new SpeechletResponseEnvelope();
-    responseEnvelope.setSessionAttributes(sessionAttributesOutput);
-
-    SpeechletResponse speechletResponse = new SpeechletResponse();
-    speechletResponse.setShouldEndSession(builder.isConversationEnd());
-
-    SimpleCard card = new SimpleCard();
-    card.setContent(builder.getRawText());
-    card.setTitle("Alexa + Complibot");
-    
-    SsmlOutputSpeech outputSpeech;
-    // Create a VoiceOutput object with the SSML content generated by the manager
-    if ((voiceInput.getMessageType() != MessageType.END_OF_CONVERSATION) && (voiceInput.getMessageType() != MessageType.STOP) && (voiceInput.getMessageType() != MessageType.CANCEL)) {
-
-      @SuppressWarnings("unchecked")
-      VoiceOutput<SsmlOutputSpeech> voiceOutput = (VoiceOutput<SsmlOutputSpeech>) VoiceMessageFactory.buildOutputMessage(builder.build(), InterfaceType.ALEXA);
-      outputSpeech = voiceOutput.getImplInstance();
-    }else{
-      outputSpeech = null;
-      card = null;
-      speechletResponse.setShouldEndSession(true);
+    QuipMetadata metadata = null;
+    try {
+      if(testFlag == null || testFlag == false){ 
+        AlexaUtils.validateAlexaRequest(request, signatureCertChainUrl, signature);
```

Comment:
This function will blow up if request is null. Our nullity check should be above here?